src/audio/copier/copier.c

```diff
@@ -1446,6 +1449,38 @@ static int copier_set_sink_fmt(struct comp_dev *dev, const void *data,
 							      &sink_fmt->sink_fmt, ipc4_gtw_none,
 							      ipc4_bidirection);
 
+	/* update corresponding sink format */
+	list_for_item(sink_list, &dev->bsink_list) {
```

Comment:
looks like this piece of code in already there in copier_params() on line 1360. Can you create a helper and reuse?

Reply:
updated with a help funciton